lib/task/import/validate/csvLanguageValidator.class.php

```diff
@@ -0,0 +1,120 @@
+    {
+        $this->setTitle(self::TITLE);
+        parent::__construct($options);
+
+        $this->languages = array_keys(sfCultureInfo::getInstance()->getLanguages());
+    }
+
+    public function reset()
+    {
+        $this->languageColumnPresent = null;
+        $this->rowsWithPipeFoundInLanguage = 0;
+        $this->rowsWithInvalidLanguage = 0;
+        $this->invalidLanguages = [];
+
+        parent::reset();
+    }
+
+    public function testRow(array $header, array $row)
+    {
+        parent::testRow($header, $row);
+        $row = $this->combineRow($header, $row);
+
+        // Set if language column is present.
+        if (!isset($this->languageColumnPresent)) {
+            $this->languageColumnPresent = array_key_exists('language', $row);
+        }
+
+        // If present check contents.
+        if ($this->languageColumnPresent) {
+            if (!empty($row['language'])) {
+                // Validate language value against AtoM.
+                $errorDetailAdded = false;
+                foreach (explode('|', $row['language']) as $value) {
+                    if (!$this->isLanguageValid(trim($value))) {
+                        if (!$errorDetailAdded) {
+                            ++$this->rowsWithInvalidLanguage;
+                            $this->testData->addDetail(implode(',', $row));
+                            $errorDetailAdded = true;
+                        }
+
+                        // Keep a list of invalid language values.
+                        if (!in_array($row['language'], $this->invalidLanguages)) {
+                            $this->invalidLanguages[] = trim($value);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    public function getTestResult()
+    {
+        if (false == $this->languageColumnPresent) {
+            // language column not present in file.
+            $this->testData->addResult(sprintf("'language' column not present in file."));
+        } else {
+            // Rows exist with invalid language.
+            if (0 < $this->rowsWithInvalidLanguage) {
+                $this->testData->setStatusError();
+                $this->testData->addResult(sprintf('Rows with invalid language values: %s', $this->rowsWithInvalidLanguage));
+            }
+
+            if (0 < $this->rowsWithInvalidLanguage) {
+                $this->testData->addResult(sprintf('Invalid language values: %s', implode(', ', $this->invalidLanguages)));
+            }
+
+            if (0 == $this->rowsWithInvalidLanguage) {
+                $this->testData->addResult(sprintf("'language' column values are all valid."));
+            }
+        }
+
+        return parent::getTestResult();
+    }
+
+    // TODO: Remove these DB accesses to a wrapper class so it's not performed in the
+    // test class itself.
+    protected function isLanguageValid(string $language)
+    {
+        if (!empty($language)) {
```

Comment:
If `$language` is an empty string, this method will return null.  I think it's better to explicitly return `false` in this case to make it clear that an empty string is considered invalid.

Reply:
good suggestion! Implemented...